src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -72,6 +78,24 @@ export class RundownComponent implements OnInit, OnDestroy, OnChanges {
       .catch(error => this.logger.data(error).error('Failed subscribing to rundown actions changes.'))
   }
 
+  public scrollViewToSegment(event: PartEvent): void {
+    const element: HTMLElement | null = document.getElementById(event.segmentId)
+    if (!element || this.isInsideViewport(element)) {
+      return
+    }
+
+    document.getElementById(event.segmentId)?.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'nearest',
+    })
+  }
+
```

Comment:
You got `document.getElementById(event.segmentId)` in constant on to the top, why you don't use it here?
```
 document.getElementById(event.segmentId)?.scrollIntoView({
      behavior: 'smooth',
      block: 'nearest',
      inline: 'nearest',
    })
```
In addition you check you have a element and is no need to be optional  document.getElementById(event.segmentId)**?**.scrollIntoView

Reply:
A good point, i don't know why i didn't change that :-D 
I have changed it to use the element instead!